openedx/core/djangoapps/content_libraries/api/serializers.py

```diff
@@ -0,0 +1,64 @@
+"""
+Serializer classes for containers
+"""
+from lxml import etree
+
+from openedx.core.djangoapps.xblock import api as xblock_api
+from openedx.core.lib.xblock_serializer.api import StaticFile, XBlockSerializer
+from openedx.core.djangoapps.content_tagging.api import TagValuesByObjectIdDict, get_all_object_tags
+
+from . import containers as container_api
+
+
+class ContainerSerializer:
+    """
+    Serializes a container (a Section, Subsection, or Unit) to OLX.
+    """
+    static_files: list[StaticFile]
+    tags: TagValuesByObjectIdDict
+
+    def __init__(self, container_metadata: container_api.ContainerMetadata):
+        self.container_metadata = container_metadata
+        self.static_files = []
+        self.tags = {}
+        olx_node = self._serialize_container(container_metadata)
+
+        self.olx_str = etree.tostring(olx_node, encoding="unicode", pretty_print=True)
+
+    def _serialize_container(self, container_metadata: container_api.ContainerMetadata) -> etree.Element:
+        """
+        Serialize the given container to OLX.
+        """
+        # Create an XML node to hold the exported data
+        container_type = container_api.ContainerType(container_metadata.container_key.container_type)
+        container_key = container_metadata.container_key
+
+        olx = etree.Element(container_type.olx_tag)
+        olx.attrib["upstream"] = str(container_key)
+        olx.attrib["upstream_version"] = str(container_metadata.draft_version_num)
```

Comment:
Is it possible to set these `upstream` and `upstream_version` on paste, rather than on copy? Because we might have other uses for this `ContainerSerializer` like import/export where we don't want to have the `upstream` attribute added/changed.